### PR TITLE
Omit Base64 padding characters from Base64 representations of secrets.

### PR DESCRIPTION
**Pull request by jww3**
### Also

Added a unit test for thorough Base64 secret masking.

### CODE REVIEW

File: src/Runner.Common/HostContext.cs [hunk 1]
```diff
@@ -82,17 +82,20 @@ public HostContext(string hostType, string logFile = null)
...
+                ValueEncoders.CommandLineArgumentEscape,
+                ValueEncoders.ExpressionStringEscape,
+                ValueEncoders.JsonStringEscape,
+                ValueEncoders.UriDataEscape,
+                ValueEncoders.XmlDataEscape,
+                ValueEncoders.TrimDoubleQuotes,
+                ValueEncoders.PowerShellPreAmpersandEscape,
+                ValueEncoders.PowerShellPostAmpersandEscape
+            };
+            _secretMasker = new SecretMasker(masks);
```
Comment (jww3) [inline src/Runner.Common/HostContext.cs:97]: treat the collection of `ValueEncoder`s as an immutable set.  (There's no existing use case for modifying it post-initialization.)

File: src/Sdk/DTLogging/Logging/SecretMasker.cs [hunk 2]
```diff
@@ -135,69 +131,6 @@ public void AddValue(String value)
             }
         }
 
-        /// <summary>
-        /// This implementation assumes no more than one thread is adding regexes, values, or encoders at any given time.
-        /// </summary>
-        public void AddValueEncoder(ValueEncoder encoder)
```
Comment (jww3) [inline src/Sdk/DTLogging/Logging/SecretMasker.cs:141]: nobody was calling this post-initialization.

File: src/Sdk/DTLogging/Logging/ValueEncoders.cs [hunk 3]
```diff
@@ -1,73 +1,97 @@
...
 using System.Security;
 using System.Text;
-using System.Text.RegularExpressions;
 using Newtonsoft.Json;
 
 namespace GitHub.DistributedTask.Logging
 {
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public delegate String ValueEncoder(String value);
+    public delegate IEnumerable<string> ValueEncoder(string value);
```
Comment (jww3) [inline src/Sdk/DTLogging/Logging/ValueEncoders.cs:12]: Allow a single ValueEncode to emit multiple variations.  (In some cases, this avoids having to recompute the same intermediate data over-and-over.)

File: src/Runner.Common/HostContext.cs
Comment (jww3) [inline src/Runner.Common/HostContext.cs:1]: TODO:  discuss the special cases called-out by @thboop, [here](https://github.com/github/c2c-actions/issues/6501#issuecomment-1491893123)

File: src/Sdk/DTExpressions2/Expressions2/Sdk/ExpressionNode.cs
Comment (TingluoHuang) [inline src/Sdk/DTExpressions2/Expressions2/Sdk/ExpressionNode.cs:1]: all changes under `src/Sdk` is shared with the service, get it merged into the service first. 😄

Comment (jww3) [reply src/Sdk/DTExpressions2/Expressions2/Sdk/ExpressionNode.cs:1]: I've got a PR open to update the service.  (see reference below)

File: src/Runner.Common/HostContext.cs
Comment (jww3) [reply src/Runner.Common/HostContext.cs:1]: Potiential solution here:  https://github.com/actions/runner/pull/2520

**Pull request closed without merging**